shell_automaton/src/mempool/mempool_state.rs

```diff
@@ -69,6 +69,71 @@ impl MempoolState {
             .get(&peer)
             .map_or(false, |p| p.seen_operations.contains(op_hash))
     }
+
+    /// Get next operation with highest priority for prevalidation.
+    pub fn next_for_prevalidation(&self) -> Option<(&OperationHash, &Operation)> {
+        self.injecting_rpc_ids
+            .iter()
+            .find_map(|(hash, _)| {
+                self.pending_operations
+                    .get(hash)
+                    .map(|content| (hash, content))
+            })
+            .or_else(|| self.pending_operations.next_for_prevalidation())
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Default, Clone)]
+pub struct MempoolPendingOperations {
+    ops: HashMap<OperationHash, Operation>,
+    consensus_ops: VecDeque<OperationHash>,
+}
+
+impl MempoolPendingOperations {
+    pub fn is_empty(&self) -> bool {
+        self.ops.is_empty()
+    }
+
+    pub fn contains_key(&self, key: &OperationHash) -> bool {
+        self.ops.contains_key(key)
+    }
+
+    pub fn get(&self, key: &OperationHash) -> Option<&Operation> {
+        self.ops.get(key)
+    }
+
+    pub fn insert(&mut self, key: OperationHash, value: Operation) {
+        let op_kind = OperationKind::from_operation_content_raw(value.data().as_ref());
+        if self.ops.insert(key.clone(), value).is_none() {
+            if op_kind.is_consensus_operation() {
+                self.consensus_ops.push_back(key);
+            }
+        }
+    }
```

Comment:
One last thing, we are accumulating clippy warnings again, if we don't solve them we are soon going to be back at the same point we were a few weeks back.

Reply:
let me fix those..